forum/views.py

```diff
@@ -1,48 +1,69 @@
-from django.shortcuts import render, redirect, HttpResponse
+from django.views import generic
+from django.views.generic import View, TemplateView, DetailView, ListView, CreateView
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login
 from .models import Forum, Post, Comment, PostVote, CommentVote
-from ..base/models import BaseModel 
-from django.contrib import messages
-
-# Create your views here.
-# Will probably have to refractor, but I am not comfortable with class based views, yet, so it may be best to get some kind of MVC and then modify
-
-# User not logged in
-def index(request):
-    print "index view"
-    return HttpResponse("Index View Sir")
-
-# Page will exist for functionality and then I would add more complicated jquery 
-def reg_log(request):
-    print "log_reg view"
-    return render("forum/log_reg.html")
-
-# Would take in intial login and registration information as well as search ultimately/ much future state
-def post(request):
-    print "post view"
-    return redirect("/main")
-
-# Specific to the user once logged in
-def main(request, user_name):
-    print "main view"
-    return HttpResponse("main View Sir")
-
-# Would be standard form or html page with some user info already populated through a api pull
-def profile(request, user_name):
-    print "profile view"
-    return HttpResponse("profile View Sir")
-
-
-# Advanced feature
-def mail(request):
-    print "mail view"
-    return HttpResponse("mail View Sir")
-
-# Advanced feature.  Would be standard form or html page with some user info already populated through a api pull
-def preferences(request):
-    print "preferences view"
-    return HttpResponse("preferences View Sir")
-
-def logout(request):
-    print 'logging out'
-    request.session.clear()
-    return redirect('/')
+from django.shortcuts import render, redirect, HttpResponse, get_object_or_404
+from .forms import RegisterForm, LoginForm
+
+
+# Can change "standard query reference from object_list to whatever as below"
+class HomePageView(ListView):
+    template_name = "forum/list.html"
+    context_object_name= "post_list"
+
+    # Throwing error on my end for comment, but view works
+    #Should be first page
+    def get_queryset(self):
+        # Not sure how to pass below...need to refractor into class createview maybe?  Sending context forward as is creates error
+        vote_count = Post.objects.first().votes.count()
+        upvotes = Post.objects.first().votes.filter(vote=1).count()
+        downvotes = Post.objects.first().votes.filter(vote=-1).count()
+        total = upvotes - downvotes
+        context = {
+            "vc": vote_count,
+            "total":total
+        }
+        print(context)
+        return Post.objects.all()
```

Comment:
What are you trying to do here exactly? You're making 4 database calls but only returning one of them.

Reply:
I was running test pulls to try and get total.  Thinking that I will run for loop that access each comment and input formula to total scores.  Seems unsafe so I may need address this manually OR I can update the model to create add function for total vote score [upvotes(+1) plus downvotes (-1s)].

Reply:
I think adding function or attribute/ property to model is good idea.  Your thoughts?

Reply:
This is not correct though. You're using `first()` which one gets a single `Post`. You already have access to `Post.objects.all()` which is a many `Post` objects. Each of those has access to `votes`, right? So I don't see why you need those extra queries since they don't really do what you want.